swap_meet/vendor.py

```diff
@@ -1,2 +1,60 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if inventory:
+            self.inventory = inventory
+        else:
+            self.inventory = []
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        try:
+            self.inventory.remove(item)
+        except ValueError:
+            print("That item is not in the inventory.")
+            return False
+        else:
+            return item
+    
+    def get_by_category(self, category):
+        items_with_category = [item for item in self.inventory if item.category == category]
+        return items_with_category
+    
+    def swap_items(self, vendor, item1, item2):
+        try: 
+            index_item1 = self.inventory.index(item1) #check for items in respective vendors inventory
+            index_item2 = vendor.inventory.index(item2)  
+        except ValueError:
+            print("Item not found")
+            return False
+        else:
+            self.inventory[index_item1] = item2
+            vendor.inventory[index_item2] = item1
+            return True
+
+    def swap_first_item(self, vendor):
+        try:
+            your_item = self.inventory[0] #check that each vendor has an item in their inventory
+            their_item = vendor.inventory[0]
+        except IndexError:
+            print("Item not found")
+            return False
+        else:
+            self.swap_items(vendor, your_item, their_item)
+            return True
+
+    def get_best_by_category(self, category):
+
+        #create a list of items with the category
+        category_items = [item for item in self.inventory if item.category == category]
```

Comment:
We could reuse `get_by_category` so we don't need to rewrite the list comprehension:
```suggestion
        category_items = self.get_by_category(category)
```